core/src/main/java/edu/wpi/grip/core/operations/composite/FilterContoursOperation.java

```diff
@@ -153,54 +158,39 @@ public void perform() {
 
 
     final MatVector inputContours = inputSocket.getValue().get().getContours();
-    final MatVector outputContours = new MatVector(inputContours.size());
-    final Mat hull = new Mat();
-
-    // Add contours from the input vector to the output vector only if they pass all of the
-    // criteria (minimum
-    // area, minimum perimeter, width, and height, etc...)
-    int filteredContourCount = 0;
-    for (int i = 0; i < inputContours.size(); i++) {
-      final Mat contour = inputContours.get(i);
-
-      final Rect bb = boundingRect(contour);
-      if (bb.width() < minWidth || bb.width() > maxWidth) {
-        continue;
-      }
-      if (bb.height() < minHeight || bb.height() > maxHeight) {
-        continue;
-      }
-
-      final double area = contourArea(contour);
-      if (area < minArea) {
-        continue;
-      }
-      if (arcLength(contour, true) < minPerimeter) {
-        continue;
-      }
+    final MatVector outputContours = JavaCppUtils.stream(inputContours)
+        .filter(contour -> sizeMatch(contour, minWidth, maxWidth, minHeight, maxHeight))
+        .filter(contour -> ratioMatch(contour, minRatio, maxRatio))
+        .filter(contour -> arcLength(contour, true) >= minPerimeter)
+        .filter(contour -> contourArea(contour) >= minArea)
+        .filter(contour -> solidityInRange(contour, minSolidity, maxSolidity))
+        .filter(contour -> inRange(contour.rows(), minVertexCount, maxVertexCount))
+        .collect(JavaCppUtils.toMatVector());
```

Comment:
Isn't this much slower?

Reply:
Which part? The stream, or the collect to `MatVector`?